zephyr/wrapper.c

```diff
@@ -198,8 +193,12 @@ int interrupt_get_irq(unsigned int irq, const char *cascade)
 
 int interrupt_register(uint32_t irq, void(*handler)(void *arg), void *arg)
 {
+#ifdef CONFIG_DYNAMIC_INTERRUPTS
 	return arch_irq_connect_dynamic(irq, 0, (void (*)(const void *))handler,
 					arg, 0);
+#else
+	return -EOPNOTSUPP;
```

Comment:
Should we log an error here so that it's obvious ?

Reply:
Yes, I also believe that we should return error here and notify the user that it must enable DYNAMIC_INTERRUPTS.